arch/power/Makefile.in

```diff
@@ -9,6 +10,7 @@ INCLUDES=
 SUFFIX=
 
 P8FLAGS=-mcpu=power8
+PPCFLAGS=-maltivec
```

Comment:
I'm trying to think if only using `-maltivec` and checking for `PPC_FEATURE_ALTIVEC` at runtime could have any unexpected side-effects. For example, examining the generated code for `adler32_vmx` I see some VSX instructions as well, like `xxlnor`:
```
$ CFLAGS='-O2 -g' ./configure && make clean && make -j$(nproc)
[...]
$ objdump -S adler32_vmx.o | grep xxlnor -C 5
 12c:   70 ff 21 39     addi    r9,r1,-144
 130:   8e e0 29 7c     lvewx   v1,r9,r28
    s2acc = vec_insert(s[1], vmx_zero(), 0);
 134:   8e f0 09 7c     lvewx   v0,r9,r30
    adacc = vec_insert(s[0], vmx_zero(), 0);
 138:   17 6d ad f1     xxlnor  vs45,vs45,vs45
 13c:   6b 33 21 10     vperm   v1,v1,v6,v13
    s2acc = vec_insert(s[1], vmx_zero(), 0);
 140:   6b 33 00 10     vperm   v0,v0,v6,v13
    while (len > 0) {
 144:   00 00 00 60     nop
```
The code was generated using GCC 8.3.0-6 from Debian 10. This happened because that GCC uses `-mcpu=power8` by default under the hood, which also implies `-mvsx`.

So my concern was that if such a binary was used on an older processor, it could cause a SIGILL. But in that case, I'd expect the code to be compiled with the proper flags for that processor, so this shouldn't happen.

Reply:
I thought about that... But I wasn't sure which CPU I should target. I wrote the original code in 2017 and back then the only PowerPC machine I had used was BeBox...

Reply:
Here it generates:

```

adler32_vmx.o:     file format elf32-powerpc


Disassembly of section .text:

00000000 <vec_hadduh>:
   0:	94 21 ff f0 	stwu    r1,-16(r1)
   4:	10 01 03 0c 	vspltisb v0,1
   8:	10 20 10 08 	vmuloub v1,v0,v2
   c:	10 40 12 08 	vmuleub v2,v0,v2
  10:	38 21 00 10 	addi    r1,r1,16
  14:	10 41 10 40 	vadduhm v2,v1,v2
  18:	4e 80 00 20 	blr
  1c:	60 00 00 00 	nop

00000020 <vec_hadduw>:
  20:	94 21 ff f0 	stwu    r1,-16(r1)
  24:	10 01 03 4c 	vspltish v0,1
  28:	10 20 10 48 	vmulouh v1,v0,v2
  2c:	10 40 12 48 	vmuleuh v2,v0,v2
  30:	38 21 00 10 	addi    r1,r1,16
  34:	10 41 10 80 	vadduwm v2,v1,v2
  38:	4e 80 00 20 	blr
  3c:	60 00 00 00 	nop

00000040 <adler32_vmx>:
  40:	2b 85 00 01 	cmplwi  cr7,r5,1
  44:	54 6a 84 3e 	rlwinm  r10,r3,16,16,31
  48:	54 63 04 3e 	clrlwi  r3,r3,16
  4c:	41 9e 03 08 	beq     cr7,354 <adler32_vmx+0x314>
  50:	2f 84 00 00 	cmpwi   cr7,r4,0
  54:	41 9e 03 48 	beq     cr7,39c <adler32_vmx+0x35c>
  58:	2b 85 00 0f 	cmplwi  cr7,r5,15
  5c:	40 9d 03 48 	ble     cr7,3a4 <adler32_vmx+0x364>
  60:	94 21 ff 10 	stwu    r1,-240(r1)
  64:	70 86 00 0f 	andi.   r6,r4,15
  68:	92 61 00 bc 	stw     r19,188(r1)
  6c:	92 81 00 c0 	stw     r20,192(r1)
  70:	92 a1 00 c4 	stw     r21,196(r1)
  74:	92 c1 00 c8 	stw     r22,200(r1)
  78:	92 e1 00 cc 	stw     r23,204(r1)
  7c:	93 01 00 d0 	stw     r24,208(r1)
  80:	93 21 00 d4 	stw     r25,212(r1)
  84:	93 41 00 d8 	stw     r26,216(r1)
  88:	93 61 00 dc 	stw     r27,220(r1)
  8c:	93 81 00 e0 	stw     r28,224(r1)
  90:	93 a1 00 e4 	stw     r29,228(r1)
  94:	93 c1 00 e8 	stw     r30,232(r1)
  98:	93 e1 00 ec 	stw     r31,236(r1)
  9c:	40 82 02 64 	bne     300 <adler32_vmx+0x2c0>
  a0:	3d 80 00 00 	lis     r12,0
  a4:	10 a0 03 8c 	vspltisw v5,0
  a8:	3f e0 00 00 	lis     r31,0
  ac:	10 c4 03 8c 	vspltisw v6,4
  b0:	39 8c 00 00 	addi    r12,r12,0
  b4:	3c 00 80 07 	lis     r0,-32761
  b8:	11 41 03 4c 	vspltish v10,1
  bc:	39 60 00 00 	li      r11,0
  c0:	55 8c 00 36 	rlwinm  r12,r12,0,0,27
  c4:	10 e1 03 0c 	vspltisb v7,1
  c8:	3b ff 00 00 	addi    r31,r31,0
  cc:	60 00 80 71 	ori     r0,r0,32881
  d0:	61 6b ff f1 	ori     r11,r11,65521
  d4:	7c c9 33 78 	mr      r9,r6
  d8:	38 e0 15 b0 	li      r7,5552
  dc:	3a a0 ff f0 	li      r21,-16
  e0:	3a c0 ff e0 	li      r22,-32
  e4:	3a e0 ff d0 	li      r23,-48
  e8:	3b 00 ff c0 	li      r24,-64
  ec:	3b 20 ff b0 	li      r25,-80
  f0:	3b 40 ff a0 	li      r26,-96
  f4:	3b 60 ff 90 	li      r27,-112
  f8:	3b 80 ff 80 	li      r28,-128
  fc:	3b a0 ff 70 	li      r29,-144
 100:	3b c0 ff 60 	li      r30,-160
 104:	60 00 00 00 	nop
 108:	60 00 00 00 	nop
 10c:	60 00 00 00 	nop
 110:	7d 07 4a 14 	add     r8,r7,r9
 114:	7f 88 28 40 	cmplw   cr7,r8,r5
 118:	40 9d 00 14 	ble     cr7,12c <adler32_vmx+0xec>
 11c:	7c e9 28 50 	subf    r7,r9,r5
 120:	2f 87 00 0f 	cmpwi   cr7,r7,15
 124:	40 9d 01 1c 	ble     cr7,240 <adler32_vmx+0x200>
 128:	7c a8 2b 78 	mr      r8,r5
 12c:	90 61 00 a0 	stw     r3,160(r1)
 130:	91 41 00 90 	stw     r10,144(r1)
 134:	38 61 00 b0 	addi    r3,r1,176
 138:	7c 00 f8 ce 	lvx     v0,0,r31
 13c:	7c ea 26 70 	srawi   r10,r7,4
 140:	7d 24 4a 14 	add     r9,r4,r9
 144:	7d 49 03 a6 	mtctr   r10
 148:	7d 63 a8 8e 	lvewx   v11,r3,r21
 14c:	7c 23 b0 8e 	lvewx   v1,r3,r22
 150:	11 65 58 2b 	vperm   v11,v5,v11,v0
 154:	10 25 08 2b 	vperm   v1,v5,v1,v0
 158:	7d 00 60 ce 	lvx     v8,0,r12
 15c:	60 00 00 00 	nop
 160:	7d a0 48 ce 	lvx     v13,0,r9
 164:	11 8b 31 84 	vslw    v12,v11,v6
 168:	39 29 00 10 	addi    r9,r9,16
 16c:	11 81 60 80 	vadduwm v12,v1,v12
 170:	11 28 6a 08 	vmuleub v9,v8,v13
 174:	10 28 68 08 	vmuloub v1,v8,v13
 178:	10 0d 38 08 	vmuloub v0,v13,v7
 17c:	11 ad 3a 08 	vmuleub v13,v13,v7
 180:	10 21 48 40 	vadduhm v1,v1,v9
 184:	10 00 68 40 	vadduhm v0,v0,v13
 188:	11 21 50 48 	vmulouh v9,v1,v10
 18c:	11 aa 00 48 	vmulouh v13,v10,v0
 190:	10 21 52 48 	vmuleuh v1,v1,v10
 194:	10 0a 02 48 	vmuleuh v0,v10,v0
 198:	10 29 08 80 	vadduwm v1,v9,v1
 19c:	10 0d 00 80 	vadduwm v0,v13,v0
 1a0:	10 21 60 80 	vadduwm v1,v1,v12
 1a4:	11 60 58 80 	vadduwm v11,v0,v11
 1a8:	42 00 ff b8 	bdnz    160 <adler32_vmx+0x120>
 1ac:	39 21 00 b0 	addi    r9,r1,176
 1b0:	55 4a 20 36 	rlwinm  r10,r10,4,0,27
 1b4:	7d 69 b9 ce 	stvx    v11,r9,r23
 1b8:	7d 09 43 78 	mr      r9,r8
 1bc:	39 01 00 b0 	addi    r8,r1,176
 1c0:	7d 68 c1 ce 	stvx    v11,r8,r24
 1c4:	7c c6 52 14 	add     r6,r6,r10
 1c8:	7f 85 48 40 	cmplw   cr7,r5,r9
 1cc:	7c 28 d9 ce 	stvx    v1,r8,r27
 1d0:	7c 28 e1 ce 	stvx    v1,r8,r28
 1d4:	7d 68 c9 ce 	stvx    v11,r8,r25
 1d8:	7c 28 e9 ce 	stvx    v1,r8,r29
 1dc:	7d 68 d1 ce 	stvx    v11,r8,r26
 1e0:	81 41 00 80 	lwz     r10,128(r1)
 1e4:	7c 28 f1 ce 	stvx    v1,r8,r30
 1e8:	80 61 00 74 	lwz     r3,116(r1)
 1ec:	82 81 00 34 	lwz     r20,52(r1)
 1f0:	81 01 00 40 	lwz     r8,64(r1)
 1f4:	82 61 00 28 	lwz     r19,40(r1)
 1f8:	7d 4a 1a 14 	add     r10,r10,r3
 1fc:	80 61 00 68 	lwz     r3,104(r1)
 200:	7d 08 a2 14 	add     r8,r8,r20
 204:	82 81 00 5c 	lwz     r20,92(r1)
 208:	7d 4a 1a 14 	add     r10,r10,r3
 20c:	80 61 00 1c 	lwz     r3,28(r1)
 210:	7d 08 9a 14 	add     r8,r8,r19
 214:	7e 8a a2 14 	add     r20,r10,r20
 218:	7d 08 1a 14 	add     r8,r8,r3
 21c:	7c 74 00 16 	mulhwu  r3,r20,r0
 220:	7d 48 00 16 	mulhwu  r10,r8,r0
 224:	54 63 8b fe 	rlwinm  r3,r3,17,15,31
 228:	55 4a 8b fe 	rlwinm  r10,r10,17,15,31
 22c:	7c 63 59 d6 	mullw   r3,r3,r11
 230:	7d 4a 59 d6 	mullw   r10,r10,r11
 234:	7c 63 a0 50 	subf    r3,r3,r20
 238:	7d 4a 40 50 	subf    r10,r10,r8
 23c:	41 9d fe d4 	bgt     cr7,110 <adler32_vmx+0xd0>
 240:	7f 86 28 40 	cmplw   cr7,r6,r5
 244:	40 9c 00 6c 	bge     cr7,2b0 <adler32_vmx+0x270>
 248:	7d 26 28 51 	subf.   r9,r6,r5
 24c:	41 82 00 34 	beq     280 <adler32_vmx+0x240>
 250:	39 04 ff ff 	addi    r8,r4,-1
 254:	38 e6 ff ff 	addi    r7,r6,-1
 258:	7c c8 32 14 	add     r6,r8,r6
 25c:	7c 84 3a 14 	add     r4,r4,r7
 260:	7d 26 4a 14 	add     r9,r6,r9
 264:	7d 24 48 50 	subf    r9,r4,r9
 268:	7d 29 03 a6 	mtctr   r9
 26c:	60 00 00 00 	nop
 270:	8d 24 00 01 	lbzu    r9,1(r4)
 274:	7c 63 4a 14 	add     r3,r3,r9
 278:	7d 4a 1a 14 	add     r10,r10,r3
 27c:	42 00 ff f4 	bdnz    270 <adler32_vmx+0x230>
 280:	3d 20 80 07 	lis     r9,-32761
 284:	38 e0 00 00 	li      r7,0
 288:	61 29 80 71 	ori     r9,r9,32881
 28c:	60 e7 ff f1 	ori     r7,r7,65521
 290:	7d 03 48 16 	mulhwu  r8,r3,r9
 294:	7d 2a 48 16 	mulhwu  r9,r10,r9
 298:	55 08 8b fe 	rlwinm  r8,r8,17,15,31
 29c:	55 29 8b fe 	rlwinm  r9,r9,17,15,31
 2a0:	7d 08 39 d6 	mullw   r8,r8,r7
 2a4:	7d 29 39 d6 	mullw   r9,r9,r7
 2a8:	7c 68 18 50 	subf    r3,r8,r3
 2ac:	7d 49 50 50 	subf    r10,r9,r10
 2b0:	55 4a 80 1e 	rlwinm  r10,r10,16,0,15
 2b4:	82 61 00 bc 	lwz     r19,188(r1)
 2b8:	82 81 00 c0 	lwz     r20,192(r1)
 2bc:	7d 43 1b 78 	or      r3,r10,r3
 2c0:	82 a1 00 c4 	lwz     r21,196(r1)
 2c4:	82 c1 00 c8 	lwz     r22,200(r1)
 2c8:	82 e1 00 cc 	lwz     r23,204(r1)
 2cc:	83 01 00 d0 	lwz     r24,208(r1)
 2d0:	83 21 00 d4 	lwz     r25,212(r1)
 2d4:	83 41 00 d8 	lwz     r26,216(r1)
 2d8:	83 61 00 dc 	lwz     r27,220(r1)
 2dc:	83 81 00 e0 	lwz     r28,224(r1)
 2e0:	83 a1 00 e4 	lwz     r29,228(r1)
 2e4:	83 c1 00 e8 	lwz     r30,232(r1)
 2e8:	83 e1 00 ec 	lwz     r31,236(r1)
 2ec:	38 21 00 f0 	addi    r1,r1,240
 2f0:	4e 80 00 20 	blr
 2f4:	60 00 00 00 	nop
 2f8:	60 00 00 00 	nop
 2fc:	60 00 00 00 	nop
 300:	20 c6 00 10 	subfic  r6,r6,16
 304:	39 24 ff ff 	addi    r9,r4,-1
 308:	7c c9 03 a6 	mtctr   r6
 30c:	60 00 00 00 	nop
 310:	8d 09 00 01 	lbzu    r8,1(r9)
 314:	7c 63 42 14 	add     r3,r3,r8
 318:	7d 4a 1a 14 	add     r10,r10,r3
 31c:	42 00 ff f4 	bdnz    310 <adler32_vmx+0x2d0>
 320:	3d 20 80 07 	lis     r9,-32761
 324:	38 e0 00 00 	li      r7,0
 328:	61 29 80 71 	ori     r9,r9,32881
 32c:	60 e7 ff f1 	ori     r7,r7,65521
 330:	7d 03 48 16 	mulhwu  r8,r3,r9
 334:	7d 2a 48 16 	mulhwu  r9,r10,r9
 338:	55 08 8b fe 	rlwinm  r8,r8,17,15,31
 33c:	55 29 8b fe 	rlwinm  r9,r9,17,15,31
 340:	7d 08 39 d6 	mullw   r8,r8,r7
 344:	7d 29 39 d6 	mullw   r9,r9,r7
 348:	7c 68 18 50 	subf    r3,r8,r3
 34c:	7d 49 50 50 	subf    r10,r9,r10
 350:	4b ff fd 50 	b       a0 <adler32_vmx+0x60>
 354:	88 c4 00 00 	lbz     r6,0(r4)
 358:	3d 20 80 07 	lis     r9,-32761
 35c:	38 e0 00 00 	li      r7,0
 360:	61 29 80 71 	ori     r9,r9,32881
 364:	60 e7 ff f1 	ori     r7,r7,65521
 368:	7c c6 1a 14 	add     r6,r6,r3
 36c:	7d 06 48 16 	mulhwu  r8,r6,r9
 370:	55 08 8b fe 	rlwinm  r8,r8,17,15,31
 374:	7c 68 39 d6 	mullw   r3,r8,r7
 378:	7d 03 30 50 	subf    r8,r3,r6
 37c:	7d 4a 42 14 	add     r10,r10,r8
 380:	7c 6a 48 16 	mulhwu  r3,r10,r9
 384:	54 63 8b fe 	rlwinm  r3,r3,17,15,31
 388:	7c 63 39 d6 	mullw   r3,r3,r7
 38c:	7c 63 50 50 	subf    r3,r3,r10
 390:	54 63 80 1e 	rlwinm  r3,r3,16,0,15
 394:	7c 63 43 78 	or      r3,r3,r8
 398:	4e 80 00 20 	blr
 39c:	38 60 00 01 	li      r3,1
 3a0:	4e 80 00 20 	blr
 3a4:	2f 85 00 00 	cmpwi   cr7,r5,0
 3a8:	41 9e 00 28 	beq     cr7,3d0 <adler32_vmx+0x390>
 3ac:	38 84 ff ff 	addi    r4,r4,-1
 3b0:	7c a9 03 a6 	mtctr   r5
 3b4:	60 00 00 00 	nop
 3b8:	60 00 00 00 	nop
 3bc:	60 00 00 00 	nop
 3c0:	8d 24 00 01 	lbzu    r9,1(r4)
 3c4:	7c 63 4a 14 	add     r3,r3,r9
 3c8:	7d 4a 1a 14 	add     r10,r10,r3
 3cc:	42 00 ff f4 	bdnz    3c0 <adler32_vmx+0x380>
 3d0:	3d 20 80 07 	lis     r9,-32761
 3d4:	38 e0 00 00 	li      r7,0
 3d8:	61 29 80 71 	ori     r9,r9,32881
 3dc:	60 e7 ff f1 	ori     r7,r7,65521
 3e0:	7d 0a 48 16 	mulhwu  r8,r10,r9
 3e4:	7d 23 48 16 	mulhwu  r9,r3,r9
 3e8:	55 08 8b fe 	rlwinm  r8,r8,17,15,31
 3ec:	55 29 8b fe 	rlwinm  r9,r9,17,15,31
 3f0:	7d 08 39 d6 	mullw   r8,r8,r7
 3f4:	7d 29 39 d6 	mullw   r9,r9,r7
 3f8:	7d 48 50 50 	subf    r10,r8,r10
 3fc:	7c 69 18 50 	subf    r3,r9,r3
 400:	55 49 80 1e 	rlwinm  r9,r10,16,0,15
 404:	7d 23 1b 78 	or      r3,r9,r3
 408:	4e 80 00 20 	blr
```

Reply:
"-maltivec" is still default, but configure changes it to "-maltivec -mno-vsx" if compiler supports both options. This avoids targeting specific processor and as such cause no conflict between 32-bit and 64-bit processors. It still assumes compiler either supports both, only "-maltivec" or neither. 

If neither is supported, relevant source files are excluded from build, so the default value has no effect.